Tests/IGListBatchUpdateDataTests.m

```diff
@@ -123,14 +123,28 @@ - (void)test_whenMovingSections_withMoveFromConflictWithDelete_thatResultDropsTh
     IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
                                                                            deleteSections:indexSet(@[@2])
                                                                              moveSections:[NSSet setWithArray:@[newMove(2, 6), newMove(0, 2)]]
-                                                                         insertIndexPaths:[NSArray new]
-                                                                         deleteIndexPaths:[NSArray new]
-                                                                           moveIndexPaths:[NSArray new]];
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[]
+                                                                           moveIndexPaths:@[]];
     XCTAssertEqual(result.deleteSections.count, 1);
     XCTAssertEqual(result.moveSections.count, 1);
     XCTAssertEqual(result.insertSections.count, 0);
     XCTAssertEqualObjects(result.deleteSections, indexSet(@[@2]));
     XCTAssertEqualObjects(result.moveSections.anyObject, newMove(0, 2));
 }
 
+- (void)test_whenDuplicateDeleteIndexPaths_thatDeletesStaggered {
+    IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
+                                                                           deleteSections:indexSet(@[])
+                                                                             moveSections:[NSSet new]
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[newPath(2, 0), newPath(2, 0), newPath(2, 1), newPath(2, 4)]
+                                                                           moveIndexPaths:@[]];
+    XCTAssertEqual(result.deleteIndexPaths.count, 4);
+
+    NSArray *expected = @[newPath(2, 0), newPath(2, 1), newPath(2, 3), newPath(2, 4)];
```

Comment:
Ya, but I _think_ that'll be ok b/c they were telling the datasource to delete {2,0} and {2,0} and updating their data source to reflect 2 deletions. So we **have** to delete 2 things, otherwise UICV will throw. And UICV will bug if we delete {2,0} 2x, so instead {2,0} {2,1} is the next-best thing.

Tho I think there might be a bug here, I think the expectation should be

```objc
NSArray *expected = @[newPath(2, 0), newPath(2, 1), newPath(2, 2), newPath(2, 4)];
```